### ISSUE TITLE
Can't load data using RPostgreSQL in an R Markdown code chunk

### ISSUE THREAD

**Issue by leehach** [rstudio/rmarkdown#663]
I'm using R Markdown in RStudio to create a beamer presentation and I am having trouble loading data from a Postgres database. The code I am using in the R Markdown chunk is:

``````
```{r, echo=FALSE, message=FALSE}
sql = "SELECT 1;"
connPostgres = dbConnect(
  dbDriver("PostgreSQL"), dbname="universe", 
  host="localhost", port=5432, 
  user="lee", password="***"
)
df = dbGetQuery(connPostgres, sql)
```
``````

(`SELECT 1;` is not a real query, but it illustrates the problem and can be tested on any Postgres install.)

When I run Knit PDF, it fails with the following message:

```
Error in file(con, "r") : cannot open the connection
Calls: <Anonymous> ... withVisible -> eval -> eval -> paste -> readLines -> file
Execution halted
```

Needless to say, I have tested the problematic statements outside of the knit process (i.e., sending them directly to the R console) and they work fine. So why won't they work in a code chunk?

As a temporary workaround, I loaded the data into a data frame and saved the workspace to an `.RData` file. Then I loaded the workspace in a code chunk. That worked, and all subsequent operations on the data frame worked.

**Comment by yihui** [comment]
Are you sure it was this chunk that caused the error? I mean, you can try to remove this chunk and see if the problem persists.

Anyway, we cannot do anything without a minimal reproducible example. Also bear in mind that when you Knit PDF, the R Markdown document is compiled in a new R session, in which the working directory is the directory of your Rmd file (not necessarily the working directory of your current R session), so it is possible that you are able to read a certain file in the current R session but cannot in a new R session.

**Comment by leehach** [comment]
This is almost a year-and-a-half old. I just tried it in a new document, and everything seems to work.

I'm pretty sure this code chunk *was* the issue, because as I explained in the last paragraph, everything else in the script worked if I replaced the RPostgreSQL call. But everything about my setup has been upgraded since then (R, rmarkdown, RPostgreSQL, …), so I wouldn't be able reproduce the exact configuration that I was running when I reported it. It works now, so the whole issue is moot.
